src/commands/project/deploy/quick.ts

```diff
@@ -109,6 +106,13 @@ export default class DeployMetadataQuick extends SfCommand<DeployResultJson> {
       return asyncFormatter.getJson();
     }
 
+    const mdapiDeploy = new MetadataApiDeploy({
+      usernameOrConnection: connection,
+      id: deployId,
+      apiOptions: {
+        rest: api === API['REST'],
+      },
+    });
     const result = await mdapiDeploy.pollStatus({
```

Comment:
The polling was reusing the `MetadataApiDeploy` instance created from the validation ID (aka jobId) so it was polling for the validation deploy status, not the `deployRecentValidation` status.